Jellyfin.Server/Migrations/Routines/CreateUserLoggingConfigFile.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using MediaBrowser.Common.Configuration;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+
+namespace Jellyfin.Server.Migrations.Routines
+{
+    /// <summary>
+    /// Migration to initialize the user logging configuration file "logging.user.json".
+    /// If the deprecated logging.json file exists and has a custom config, it will be used as logging.user.json,
+    /// otherwise a blank file will be created.
+    /// </summary>
+    internal class CreateUserLoggingConfigFile : IMigrationRoutine
+    {
+        /// <summary>
+        /// An empty logging JSON configuration, which will be used as the default contents for the user settings config file.
+        /// </summary>
+        private const string EmptyLoggingConfig = @"{ ""Serilog"": { } }";
+
+        /// <summary>
+        /// File history for logging.json as existed during this migration creation. The contents for each has been minified.
+        /// </summary>
+        private readonly List<string> _defaultConfigHistory = new List<string>
+        {
+            // 9a6c27947353585391e211aa88b925f81e8cd7b9
+            @"{""Serilog"":{""MinimumLevel"":{""Default"":""Information"",""Override"":{""Microsoft"":""Warning"",""System"":""Warning""}},""WriteTo"":[{""Name"":""Console"",""Args"":{""outputTemplate"":""[{Timestamp:HH:mm:ss}] [{Level:u3}] [{ThreadId}] {SourceContext}: {Message:lj}{NewLine}{Exception}""}},{""Name"":""Async"",""Args"":{""configure"":[{""Name"":""File"",""Args"":{""path"":""%JELLYFIN_LOG_DIR%//log_.log"",""rollingInterval"":""Day"",""retainedFileCountLimit"":3,""rollOnFileSizeLimit"":true,""fileSizeLimitBytes"":100000000,""outputTemplate"":""[{Timestamp:yyyy-MM-dd HH:mm:ss.fff zzz}] [{Level:u3}] [{ThreadId}] {SourceContext}: {Message}{NewLine}{Exception}""}}]}}],""Enrich"":[""FromLogContext"",""WithThreadId""]}}",
```

Comment:
A hash might be more readable than this, but that's not important.

Reply:
The json string is used specifically because we are not comparing by a hash of the file contents but doing a deep equality check of the json data. This will take care of matching files that only differ in whitespace etc.